### PR TITLE
Added workspace_name label in namespace template

### PR DESCRIPTION
**Pull request by bamachrn**
This PR adds all workspace_name label to all the appstudio namespaces.

Paired with https://github.com/codeready-toolchain/toolchain-e2e/pull/721

### CODE REVIEW

File: deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml [hunk 1]
```diff
@@ -14,6 +14,8 @@ objects:
       name: ${USERNAME}-tenant
       toolchain.dev.openshift.com/workspace: ${USERNAME}
       argocd.argoproj.io/managed-by: gitops-service-argocd
+      #billing labels ref: https://github.com/redhat-appstudio/book/blob/main/ADR/0010-namespace-metadata.md
+      appstudio.redhat.com/workspace_name: ${USERNAME}
```
[the remark this answers is not in the document]
Comment (bamachrn) [reply deploy/templates/nstemplatetiers/appstudio/ns_tenant.yaml:18]: thanks for the pointer. I have now added it to sub-space and test template

File: test/templates/nstemplatetiers/appstudio/ns_tenant.yaml [hunk 2]
```diff
@@ -13,6 +13,7 @@ objects:
     labels:
       name: ${USERNAME}-tenant
       toolchain.dev.openshift.com/workspace: ${USERNAME}
+      appstudio.redhat.com/workspace_name: ${USERNAME}
```
Comment (MatousJobanek) [inline test/templates/nstemplatetiers/appstudio/ns_tenant.yaml:16]: there is no need to update also the test template - this is just a dummy thing.

File: deploy/templates/nstemplatetiers/appstudio-env/ns_env.yaml [hunk 3]
```diff
@@ -14,6 +14,7 @@ objects:
       name: ${USERNAME}-env
       toolchain.dev.openshift.com/workspace: ${USERNAME}
       argocd.argoproj.io/managed-by: gitops-service-argocd
+      appstudio.redhat.com/workspace_name: ${USERNAME}
```
Comment (MatousJobanek) [inline deploy/templates/nstemplatetiers/appstudio-env/ns_env.yaml:17]:
```suggestion
      #billing labels ref: https://github.com/redhat-appstudio/book/blob/main/ADR/0010-namespace-metadata.md
      appstudio.redhat.com/workspace_name: ${USERNAME}
```

File: test/templates/nstemplatetiers/appstudio/ns_tenant.yaml (hunk 2 above)
Comment (bamachrn) [reply test/templates/nstemplatetiers/appstudio/ns_tenant.yaml:16]: ack

Comment (bamachrn) [reply test/templates/nstemplatetiers/appstudio/ns_tenant.yaml:16]: ok, let me remove this. we dont need the billing labels everywhere.

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**